firmware/Core/Src/telecommands/telecommand_definitions.c

```diff
@@ -388,9 +392,37 @@ uint8_t TCMDEXEC_heartbeat_on(const char *args_str, TCMD_TelecommandChannel_enum
 
 uint8_t TCMDEXEC_core_system_stats(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
                         char *response_output_buf, uint16_t response_output_buf_len) {
-    // TODO: implement this (Issue #103)
-    // Use `TCMD_get_agenda_used_slots_count`
-    snprintf(response_output_buf, response_output_buf_len, "System stats: TODO\n");
+    
+    uint32_t epoch = TIM_get_current_system_uptime_ms(); 
+
+    uint32_t last_resync_ms = TIM_get_system_uptime_at_last_time_resync_ms(); // timestamp at sync
+
+    uint32_t uptime = TIM_get_current_system_uptime_ms(); //current system uptime   
+
+    uint32_t delta_uptime = uptime - last_resync_ms; // time between last time sync 
+
+    int16_t latest_tcmd_sent = TCMD_get_most_recent_tcmd_timestamp_sent();
+
+    uint32_t cmd_count = TCMD_get_tcmd_count();
+
+    uint8_t mount_status = LFS_get_mount_status();
+
+    //TODO: 
+    //  logging information for beacon count and temperature still required 
+
+    //  Getting information to convert epoch into a readable timestamp
+    time_t seconds = (time_t)(epoch/ 1000U);
+    uint16_t ms = epoch - 1000U * seconds;
```

Comment:
`ms` isn't really an acceptable variable name. The modulus operator is the correct way to extract this value.